daffodil-japi/src/main/scala/org/apache/daffodil/japi/Daffodil.scala

```diff
@@ -514,6 +507,11 @@ class DataProcessor private[japi] (private var dp: SDataProcessor)
    */
   def save(output: WritableByteChannel): Unit = dp.save(output)
 
+  def walkRuntimeSchema(handler: RuntimeSchemaHandler): Unit = {
+    val walker = new RuntimeSchemaWalker(dp)
```

Comment:
I think we already have a couple different walkers. We have an` AbsractDSOMWalker` that is used by the experimental NiFi Daffodil Record processor. We also have the walker used by the C Code generator. I think they walk different things. Does this also walk something different? Is there any chance these could all be combined? Seems we might be doing something wrong if we have a bunch of different ways to walk different things?

Reply:
Right, so I think that decision, to use a DSOM walker for NiFi, was a mistake. 

When we are interfacing Daffodil's runtime-1 output to a different data fabric, it makes sense to use the metadata structures of runtime-1 rather than DSOM. Everything we need seems to be available on the RuntimeData objects. We don't need the complexity of DSOM.